### ISSUE TITLE
[Issue]  Request to support macOS x64

### ISSUE THREAD

**Issue by csukuangfj** [ml-explore/mlx#1201]
**Describe the issue**

mlx supports both macOS arm64 and Linux x64. However, it does not support macOS x64.

I get the following error when building mlx on macOS x64. As a result, I could not test the changes I made to mlx on my macOS x64 computer.

```
-- Building MLX for x86_64 processor on Darwin
CMake Error at CMakeLists.txt:39 (message):
  Building for x86_64 on macOS is not supported.  If you are on an Apple
  silicon system, check the build documentation for possible fixes:
  https://ml-explore.github.io/mlx/build/html/install.html#build-from-source


-- Configuring incomplete, errors occurred!
```

**Comment by awni** [comment]
There's a [cmake flag](https://github.com/ml-explore/mlx/blob/main/CMakeLists.txt#L20) `MLX_ENABLE_X64_MAC` for that which is default to error. It should build for x86 if you set the flag, but we don't test on x86 and can't promise it works. Though if you encounter any issues and want to send a fix that is great.

**Comment by csukuangfj** [comment]
> There's a [cmake flag](https://github.com/ml-explore/mlx/blob/main/CMakeLists.txt#L20) MLX_ENABLE_X64_MAC for that which is default to OFF

Thanks!

Is it possible that it is set to ON automatically when compiled on macOS x64?

**Comment by awni** [comment]
No, we have it off by default because in general we don't want to encourage using MLX on x86. The problem is that sometimes people are using rosetta on an Arm machine so we like to fail in those situations.
